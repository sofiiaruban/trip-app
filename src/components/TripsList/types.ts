export interface TripProps {
  cityImgSrc: string
  cityName: string
  startDate: Date
  endDate: Date
}
