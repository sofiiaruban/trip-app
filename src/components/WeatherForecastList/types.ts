export interface WeatherForecastItemProps {
  date: string
  iconSrc: string
  iconName: string
  degree: string
}

interface HourlyWeather {
  datetime: string
  datetimeEpoch: number
  temp: number
  feelslike: number
  humidity: number
  dew: number
  precip: number
  precipprob: number
  snow: number
  snowdepth: number
  preciptype: string | null
  windgust: number
  windspeed: number
  winddir: number
  pressure: number
  visibility: number
  cloudcover: number
  solarradiation: number
  solarenergy: number
  uvindex: number
  severerisk: number
  conditions: string
  icon: string
  stations: string[] | null
  source: string
}

export interface DailyWeather extends HourlyWeather {
  sunrise: string
  sunriseEpoch: number
  sunset: string
  sunsetEpoch: number
  moonphase: number
  description: string
}
