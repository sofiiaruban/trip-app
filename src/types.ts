export interface Trip {
  cityImgSrc: string
  cityName: string
  startDate: Date
  endDate: Date
}


