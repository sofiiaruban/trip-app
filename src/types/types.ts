export interface Trip {
  cityImgSrc: string
  cityName: string
  startDate: Date
  endDate: Date
}

interface Hour {
  datetime: string
  datetimeEpoch: number
  temp: number
  feelslike: number
  humidity: number
  dew: number
  precip: number
  precipprob: number
  snow: number
  snowdepth: number
  preciptype: string[]
  windgust: number
  windspeed: number
  winddir: number
  pressure: number
  visibility: number | null
  cloudcover: number
  solarradiation: number
  solarenergy: number
  uvindex: number
  severerisk: number
  conditions: string
  icon: string
  stations: string[]
  source: string
}

export interface Day {
  datetime: string
  datetimeEpoch: number
  tempmax: number
  tempmin: number
  temp: number
  feelslikemax: number
  feelslikemin: number
  feelslike: number
  dew: number
  humidity: number
  precip: number
  precipprob: number
  precipcover: number
  preciptype: string[]
  snow: number
  snowdepth: number
  windgust: number
  windspeed: number
  winddir: number
  pressure: number
  cloudcover: number
  visibility: number
  solarradiation: number
  solarenergy: number
  uvindex: number
  severerisk: number
  sunrise: string
  sunriseEpoch: number
  sunset: string
  sunsetEpoch: number
  moonphase: number
  conditions: string
  description: string
  icon: string
  stations: null | string[]
  source: string
  hours?: Hour[]
}
interface Stations {
  [key: string]: {
    distance: number
    latitude: number
    longitude: number
    useCount: number
    id: string
    name: string
    quality: number
    contribution: number
  }
}

interface CurrentConditions {
  datetime: string
  datetimeEpoch: number
  temp: number
  feelslike: number
  humidity: number
  dew: number
  precip: number
  precipprob: number
  snow: number
  snowdepth: number
  preciptype: string[]
  windgust: number
  windspeed: number
  winddir: number
  pressure: number
  visibility: number | null
  cloudcover: number
  solarradiation: number
  solarenergy: number
  uvindex: number
  conditions: string
  icon: string
  stations: string[]
  source: string
  sunrise: string
  sunriseEpoch: number
  sunset: string
  sunsetEpoch: number
  moonphase: number
}

export interface WeatherData {
  queryCost: number
  latitude: number
  longitude: number
  resolvedAddress: string
  address: string
  timezone: string
  tzoffset: number
  description?: string
  days: Day[]
  alerts?: unknown[]
  stations?: Stations
  currentConditions?: CurrentConditions
}
