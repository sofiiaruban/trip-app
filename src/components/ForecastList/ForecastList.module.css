.list {
  display: flex;
  gap: 2rem;
  list-style-type: none;
  width: 85%;
  padding: 0;
  margin: 0;
  overflow: auto;
}
