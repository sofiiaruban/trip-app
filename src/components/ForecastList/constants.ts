export enum ForecastCardType {
  SMALL = 'small',
  BIG = 'big'
}
