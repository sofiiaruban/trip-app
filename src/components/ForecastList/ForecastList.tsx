import { FC } from 'react'
import ForecastCard from './ForecastCard/ForecastCard'
import styles from './WeatherForecastList.module.css'
import { Day } from '@app/types/types'
import { ForecastCardType } from './constants'

interface ForecastListProps {
  forecastList: Day[]
}

const ForecastList: FC<ForecastListProps> = ({ forecastList }) => {
  return (
    <ul className={styles.list}>
      {forecastList.map((item) => (
        <li key={item.datetime}>
          <ForecastCard
            date={item.datetime}
            iconSrc={`@app/assets/${item.icon}.svg`}
            iconName={item.icon}
            degree={`${item.temp}/${item.feelslike}`}
            type={ForecastCardType.SMALL}
          />
        </li>
      ))}
    </ul>
  )
}

export default ForecastList
