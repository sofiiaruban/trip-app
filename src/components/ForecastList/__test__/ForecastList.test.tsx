import { render } from '@testing-library/react'
import ForecastList from '../ForecastList'
import { Day } from '@app/types/types';

const mockForecastList : Day[] = [
  {
    datetime: '2024-02-18',
    datetimeEpoch: 1645119600,
    tempmax: 30,
    tempmin: 20,
    temp: 25,
    feelslikemax: 28,
    feelslikemin: 18,
    feelslike: 22,
    dew: 10,
    humidity: 70,
    precip: 0,
    precipprob: 0,
    precipcover: 0,
    preciptype: ['none'],
    snow: 0,
    snowdepth: 0,
    windgust: 15,
    windspeed: 10,
    winddir: 180,
    pressure: 1015,
    cloudcover: 20,
    visibility: 10,
    solarradiation: 500,
    solarenergy: 2000,
    uvindex: 5,
    severerisk: 0,
    sunrise: '06:00',
    sunriseEpoch: 1645090800,
    sunset: '18:00',
    sunsetEpoch: 1645134000,
    moonphase: 0.5,
    conditions: 'Clear',
    description: 'Clear skies',
    icon: 'clear',
    stations: null,
    source: 'mock-source',
  },
];

test('ForecastList matches snapshot', () => {
  const { container } = render(<ForecastList forecastList={mockForecastList} />)

  expect(container).toMatchSnapshot()
})
