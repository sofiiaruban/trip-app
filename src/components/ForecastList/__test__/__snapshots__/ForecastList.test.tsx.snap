// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastList matches snapshot 1`] = `
<div>
  <ul
    class="list"
  >
    <li>
      <article
        class="card small"
      >
        <p
          class="day-small"
        >
          Sunday
        </p>
        <div
          class="weatherInfo-small"
        >
          <img
            alt="clear"
            class="img-small"
            src="src/assets/clear.svg"
          />
          <p
            class="degree-small"
          >
            25/22
             
            <span>
              Â°F
            </span>
          </p>
        </div>
      </article>
    </li>
  </ul>
</div>
`;
